mlflow/server/handlers.py

```diff
@@ -2355,8 +2360,18 @@ def _graphql():
     variables = request_json.get("variables")
     operation_name = request_json.get("operationName")
 
-    # Executing the GraphQL query using the Graphene schema
-    result = schema.execute(query, variables=variables, operation_name=operation_name)
+    node = parse(query)
+    field_count = count_total_field_calls(node)
+    if field_count != 1:
```

Comment:
q: do we need to be that restrictive? One of the benefits of GQL is the ability to retrieve multiple resources in one request, and this validation disables the usage. I wonder if we should set the limit to a certain number > 1, say 10.

Reply:
Do you have any specific use case? Cuz currently there's no such request from MLflow UI.
And from the report 20 requests will make the server broken, I'm a bit hesitate to limit it to 10 🤔

Reply:
I tried to see some reference, the default max root fields of Apollo router is actually 20
https://www.apollographql.com/docs/graphos/routing/security/request-limits

Reply:
Thanks for the info! In that case I'll make this configurable so users who implement it with MLflow server has responsibility to make sure it's safe. I'll set the default to 5 for now then :)